src/kili/graphql/__init__.py

```diff
@@ -0,0 +1,152 @@
+"""
+GraphQL module
+"""
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Iterable, List, NamedTuple, Optional, Type
+
+from tqdm import tqdm
+
+from kili.helpers import format_result, fragment_builder
+from kili.utils.pagination import api_throttle
+
+from ..graphql_client import GraphQLClient
+
+
+class QueryOptions(NamedTuple):
+    """Options when calling GraphQLQuery from the SDK"""
+
+    first: Optional[int] = None
+    skip: int = 0
+    disable_tqdm: bool = True
```

Comment:
you can make `disable_tqdm` an `Optional[bool]`. `None` is accepted by tqdm, this disables tqdm when the code is run in non interactive mode (which should be our default btw)

Reply:
For inner queries, we should set disable_tqdm at True by default, and only activate it when called in a client function.
I set True by default here because we tend to increase the number of inner calls to the backend and like this we don't have to always import the QueryOptions class and disable tqdm

Reply:
But that mean that for client functions, disable_tqdm should be set to None and to False ?

Reply:
| The user is not interested by the fact that we do inner calls to Kili isn't it ?
we agree on that, this is where the `disable_tqdm` should be hard-coded to `True`.
I am not really annoyed by hard-coding it to true it each time we query 1 project. And I persist that in the other cases we should propagate what the user has decided (i.e. in the "main" query to Kili).

Reply:
the reason I prefer explicit arguments in inner code is that if we change the default value one day for some reason, it will have side effects everywhere.